public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { PureComponent, ChangeEvent } from 'react';
+import { DataSourcePluginOptionsEditorProps, DataSourceSettings, SelectableValue } from '@grafana/data';
+import { DataSourceHttpSettings, FormLabel, Input, SecretFormField, Select } from '@grafana/ui';
+import { InfluxOptions, InfluxSecureJsonData } from '../types';
+
+const httpModes = [{ label: 'GET', value: 'GET' }, { label: 'POST', value: 'POST' }];
+
+export type Props = DataSourcePluginOptionsEditorProps<InfluxOptions>;
+
+type InfluxDataSourceSettings = DataSourceSettings<InfluxOptions, InfluxSecureJsonData>;
+
+export interface State {
+  config: InfluxDataSourceSettings;
+}
+
+export class ConfigEditor extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { options } = this.props;
+
+    this.state = {
+      config: ConfigEditor.defaults(options),
+    };
+  }
+
+  static getDerivedStateFromProps(props: Props, state: State) {
+    return {
+      ...state,
+      config: ConfigEditor.defaults(props.options),
+    };
+  }
+
+  static defaults = (options: InfluxDataSourceSettings) => {
+    options.secureJsonData = options.secureJsonData || {};
```

Comment:
do we really want/need a default empty `secureJsonData` object?  that seems to lead to all kinds of other issues -- like needing to keep track of all the keys...

IIUC, with `secureJsonData` -- you only include it and send the key of values you want to change.  in props you will get a list of values that exist, but never the real value